src/util/psbt/mod.rs

```diff
@@ -210,8 +214,221 @@ impl PartiallySignedTransaction {
 
         Ok(())
     }
+
+    /// Signs the input at `input_index` with private key `sk`.
+    pub fn sign<C: Signing>(&mut self, sk: &PrivateKey, input_index: usize, secp: &Secp256k1<C>) -> Result<(), SighashError> {
```

Comment:
Are we adding a separate method for signing for P2TR or extending this one?

Reply:
Oh boy, I forgot to mention that this PR does not handle taproot. Thanks for raising this. I have no idea how we will handle taproot, I had not thought about it at all. This code is basically lifted from miniscript (after reading and playing with the same code in bdk).